doc/api/n-api.md

```diff
@@ -3580,10 +3582,10 @@ if (asynchronous_action_succeeded) {
 } else {
   status = napi_reject_deferred(env, deferred, undefined);
 }
-if (status != napi_ok) return NULL;
+if (status != napi_ok) return nullptr;
 
 // At this point the deferred has been freed, so we should assign NULL to it.
-deferred = NULL;
+deferred = nullptr;
```

Comment:
Hm – I think this was intentionally C code? At least that’s what the ```` ```c ```` seems to hint at…

Reply:
But the other examples have ```` ```C ```` and use `nullptr`.  I'm assuming `C` and `c` are the same?

Reply:
Maybe change them all to ```` ```C++ ````? (Yes, [it's valid](https://github.com/github/linguist/blob/e4b94300242651173dad0e6d32cde37c8c737804/lib/linguist/languages.yml#L510).)

Reply:
I think @addaleax's point is that it is mixing c and c++ code?

Reply:
@lpinca yes, but the other examples that are labeled as `C` use `nullptr` in this file as well, so we're using `C++` most of the time anyways. I don't care what we use in the documentation, but it should be consistent throughout the file.

Reply:
The API must be usable from C, I'm wondering if we should make the examples C code to avoid any confusion over that but don't feel super strongly about that.  Either way we should be consistent.

Reply:
@fhinkel if you want to restrict this PR to the original signature change I'm happy to take on going through to make sure we as consistent throughout the doc.